Hi @Gen5551 
I think that your commit break the compilation in other kernels... see
https://github.com/Mange/rtl8192eu-linux-driver/actions/runs/22559925921/job/66344799975?pr=359

`/home/runner/work/rtl8192eu-linux-driver/rtl8192eu-linux-driver/os_dep/linux/ioctl_linux.c:28:12: error: conflicting types for ‘mac_pton’; have ‘int(const char *, u8 *)’ {aka ‘int(const char *, unsigned char *)’}
`

Algo I get some errors with gcc-14 see
https://github.com/Mange/rtl8192eu-linux-driver/actions/runs/22559925921/job/66344799939?pr=359

`8192eu.o: error: objtool: mp_xmit_packet_thread+0x1a5: thread_exit() missing __noreturn in .c/.h or NORETURN() in noreturns.h
`

Please check the two additional commits at #360 if that made sense for you please add the code to your PR and I'll merge it (and close #360)

--- a/include/osdep_service.h
+++ b/include/osdep_service.h
@@ -315,7 +315,7 @@ static __inline void thread_enter(char *name)
 	allow_signal(SIGTERM);
 #endif
 }
-void thread_exit(_completion *comp);
+__noreturn void thread_exit(_completion *comp);
 void _rtw_init_completion(_completion *comp);
 void _rtw_wait_for_comp_timeout(_completion *comp);
 void _rtw_wait_for_comp(_completion *comp);

--- a/os_dep/linux/ioctl_linux.c
+++ b/os_dep/linux/ioctl_linux.c
@@ -28,6 +28,7 @@
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 0))
 extern int mac_pton(const char *s, u8 *addr);
 #endif
+#endif
 
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 27))
 #define  iwe_stream_add_event(a, b, c, d, e)  iwe_stream_add_event(b, c, d, e)

--- a/os_dep/linux/ioctl_linux.c
+++ b/os_dep/linux/ioctl_linux.c
@@ -25,6 +25,7 @@
 #endif
 
 #ifdef __KERNEL__
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 0))
 extern int mac_pton(const char *s, u8 *addr);
 #endif
 

